src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -0,0 +1,377 @@
+    char mLocationNameBuffer[kLocationNameMaxSize + 1];
+};
+
+/**
+ * This class wraps the MapStruct object and provides a more user-friendly interface for the data.
+ */
+struct MapStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::MapStruct::Type
+{
+    MapStructureWrapper() { Set(0, CharSpan()); }
+
+    /**
+     * @brief This is a full constructor that initializes the map object with the given values. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note If aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    MapStructureWrapper(uint8_t aMapId, const CharSpan & aMapName) { Set(aMapId, aMapName); }
+
+    /**
+     * @brief This is a copy constructor that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper(const MapStructureWrapper & aOther) { *this = aOther; }
+
+    /**
+     * @brief This is an assignment operator that initializes the map object with the values from another map object. All values are
+     * deep copied.
+     * @param[in] aOther The map object to copy.
+     */
+    MapStructureWrapper & operator=(const MapStructureWrapper & aOther)
+    {
+        Set(aOther.mapID, aOther.name);
+        return *this;
+    }
+
+    /**
+     * @brief Set all fields of the map object. All values are deep copied.
+     * @param[in] aMapId The identifier of this map.
+     * @param[in] aMapName A human readable name (should not be empty string).
+     *
+     * @note Requirements regarding what combinations of fields and values are 'valid' are not checked by this class.
+     * @note if aMapName is larger than kMapNameMaxSize, it will be truncated.
+     */
+    void Set(uint8_t aMapId, const CharSpan & aMapName)
+    {
+        mapID = aMapId;
+
+        // Save the map name to the buffer so that access and lifetime is controlled.
+        // Create a CharSpan reference the text.
+        // Make sure there is always a terminating character, mainly so we can get c_str for logging.
+        memset(mMapNameBuffer, 0, sizeof(mMapNameBuffer));
+
+        if (aMapName.empty())
+        {
+            name              = CharSpan(mMapNameBuffer, 0);
+            mMapNameBuffer[0] = '\0';
+        }
+        else if (aMapName.size() > sizeof(mMapNameBuffer))
+        {
+            // Save the truncated name that fits into available size.
+            memcpy(mMapNameBuffer, aMapName.data(), sizeof(mMapNameBuffer));
+            name                            = CharSpan(mMapNameBuffer, sizeof(mMapNameBuffer));
+            mMapNameBuffer[kMapNameMaxSize] = '\0';
```

Comment:
This has all the same null-termination bugs as above.  Please don't do that.

Reply:
Fixed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/1054f736563033c4f58db3c31fea9d7a26e02df5).